internal/guest/runtime/hcsv2/workload_container.go

```diff
@@ -45,6 +46,34 @@ func updateSandboxMounts(sbid string, spec *oci.Spec) error {
 	return nil
 }
 
+func updateHugePageMounts(ctx context.Context, sbid string, spec *oci.Spec) error {
+	mountPrefix := "hugepages://"
+	for i, m := range spec.Mounts {
+		if strings.HasPrefix(m.Source, mountPrefix) {
+			mountsDir := getSandboxRootDir(sbid)
+			subPath := strings.TrimPrefix(m.Source, mountPrefix)
+			hugePageMountSource := filepath.Join(mountsDir, subPath)
+
+			spec.Mounts[i].Source = hugePageMountSource
+
+			_, err := os.Stat(hugePageMountSource)
+			if os.IsNotExist(err) {
+				if err := os.MkdirAll(hugePageMountSource, 0755); err != nil {
+					return err
+				}
+
+				// Mount the hugepage drive.
+				output, err := exec.Command("mount", "-t", "hugetlbfs", "-o", "pagesize="+hugePageMountSource, "none", hugePageMountSource).CombinedOutput()
```

Comment:
I'd prefer if we use the syscall directly (unless something was prohibiting this. `unix.Mount`)

Reply:
done.